js/mod/recent_posts.js

```diff
@@ -0,0 +1,117 @@
+                if (posts.length > 0) {
+                    const totalPosts = Array.from(document.body.querySelectorAll(".post-wrapper"))
+                    for (const post of posts) {
+                        lastPost.prepend(post.element)
+                        totalPosts.unshift(post.element)
+                    }
+
+                    while (totalPosts.length > kRecentXPosts) {
+                        document.body.removeChild(totalPosts.pop())
+                    }
+
+                    updatePageNext()
+                    makeIcon("reply")
+                    liveUpdateFaviconChanged = true
+                }
+            }
+        }
+
+        const pageNext = Array.prototype.find.apply(document.body.children, [ el => el.nodeName == "A" && el.href.startsWith(`${location.origin}/mod.php?/recent/${kRecentXPosts}&last=`) ])
+        const updatePageNext = () => {
+            const posts = document.body.querySelectorAll(".post-wrapper")
+            const oldestPost = posts[posts.length-1]
+            pageNext.href = `/mod.php?/recent/${kRecentXPosts}&last=${getPostTimestamp(oldestPost).getTime() / 1000}`
+        }
+
+        const createCheckbox = () => {
+            const id = "jon-liveposts-enabled"
+            const div = document.createElement("div")
+            const label = document.createElement("label")
+            const checkbox = document.createElement("input")
+            checkbox.type = "checkbox"
+            checkbox.id = id
+            label.innerText = "live update: "
+            label.for = id
+            div.style.display = "inline"
+            div.append(" ")
+            div.appendChild(label)
+            div.appendChild(checkbox)
+            div.append(" ")
+            return { checkbox, label, div }
+        }
+
+        const pageNums = Array.prototype.find.apply(document.body.children, [ el => el.nodeName == "P" ])
+        const form = createCheckbox()
+
+        form.checkbox.addEventListener("click", () => {
+            setTimeout(() => {
+                if (form.checkbox.checked) {
+                    liveUpdateEnabled = true
+                    liveUpdateAbortController = new AbortController()
+                    liveUpdateIntervalId = setInterval(() => updateRecentPosts(), kPullEveryX)
+                } else {
+                    liveUpdateEnabled = false
+                    clearInterval(liveUpdateIntervalId)
+                    liveUpdateAbortController.abort()
+                    liveUpdateIntervalId = null
+                    liveUpdateAbortController = null
+                }
+            }, 700)
+        })
+
+        document.body.addEventListener("mousemove", () => {
+            if (liveUpdateFaviconChanged) {
+                liveUpdateFaviconChanged = false
+                makeIcon(false)
+            }
+        })
+
+        pageNums.append("|")
+        pageNums.append(form.div)
+    }
+})()
```

Comment:
Need to wrap the whole thing in something like document.addEventListener("DOMContentLoaded", main) if you are expecting content to be present on the page (otherwise pageNums is undefined).